api/v1beta1/azuremachine_types.go

```diff
@@ -204,7 +204,7 @@ type AzureMachineStatus struct {
 	// can be added as events to the Machine object and/or logged in the
 	// controller's output.
 	// +optional
-	FailureReason *errors.MachineStatusError `json:"failureReason,omitempty"`
+	FailureReason *errors.MachinePoolStatusFailure `json:"failureReason,omitempty"`
```

Comment:
I think you want to leave `AzureMachine.Status.FailureReason` alone since `MachineStatusError` would be the correct error type. The reason we want to change it for MachinePools is because MachinePools also has its own corresponding error type.

Reply:
Do you mean I should revert to `FailureReason *errors.MachineStatusError `json:"failureReason,omitempty"`` and `func (m *MachineScope) SetFailureReason(v capierrors.MachineStatusError) {` and keep other changes? Thanks!

Reply:
That's correct @Tea4menono . 
This file defines the "types" for `AzureMachine` type resource. Notice the file name `api/v1beta1/azuremachine_types.go`.
We don't really want to change the `AzureMachine` or `Machine` types. We want the updates to `AzureMachinePool` or `MachinePool` types.

Reply:
Yep, what @nawazkh is correct.